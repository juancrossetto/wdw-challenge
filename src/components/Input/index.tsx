import React, { FC, InputHTMLAttributes } from 'react';
import './input.scss';

interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
  children?: JSX.Element | string;
}

const Input: FC<InputProps> = ({ children, ...props }) => {
  return (
    <input type={props.type} {...props} onChange={props.onChange} className={`custom-input ${props.className}`}>
      {children}
    </input>
  );
};

export default Input;
