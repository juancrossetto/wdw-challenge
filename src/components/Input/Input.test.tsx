import React from 'react';
import renderer from 'react-test-renderer';

import Input from '.';

describe('Input component', () => {
  test('Input render correctly', () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Input with value and placehonder render correctly', () => {
    const tree = renderer.create(<Input value={'some value'} placeholder="some placeholder" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
