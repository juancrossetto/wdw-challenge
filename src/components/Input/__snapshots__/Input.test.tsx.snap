// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Input render correctly 1`] = `
<input
  className="custom-input undefined"
/>
`;

exports[`Input component Input with value and placehonder render correctly 1`] = `
<input
  className="custom-input undefined"
  placeholder="some placeholder"
  value="some value"
/>
`;
